src/UI/Titlebar.re

```diff
@@ -44,15 +45,18 @@ module Styles = {
   ];
 };
 
-let make = (~isFocused, ~isMaximized, ~title, ~theme, ~font: UiFont.t, ()) =>
+let make =
+    (~isFocused, ~isMaximized, ~title, ~theme, ~window, ~font: UiFont.t, ()) =>
   if (isMaximized) {
     React.empty;
   } else {
     switch (Revery.Environment.os) {
     | Mac =>
-      <View style={Styles.container(~isFocused, ~theme)}>
+      <Clickable
+        onDoubleClick={_ => Revery.Window.maximize(window)}
```

Comment:
Actually, I don't think we need to put `window` in the state to implement this either. If the call to `maximize` is moved to a store, the window could just be injected into that store.

The UI shouldn't be causing side-effects anyway, as triggering side-effects anywhere and everywhere tends to make the logic very hard to follow. The UI should only dispatch `action`s (which should be named `event` or `msg` to better convey its actual meaning). The `action` should then be handled by a store (in the old architecture) or `update` function (in the new Feature-based architecture) which will update the `state` and/or trigger side-effects through an `Isolinear.Effect`. This is (approximately) what's usually called [The Elm Architecture](https://guide.elm-lang.org/architecture/).

Reply:
Here, for example, you can see `setTitle` being injected into the `TitleStore`:

https://github.com/onivim/oni2/blob/ea560d1d9c4e78d59028f83ca81c7a7d28a54d63/src/Store/TitleStoreConnector.re#L104

You could either pass `window` in as an argument to `start` here as well, or perhaps a `maximize` function to make the store easier to test without creating or faking a window instance.

Reply:
Gotcha, thank you so much for the explanation. Reworking my code now!

Reply:
I think I managed to figure it out in my latest commit. Let me know if there is still anything that needs to be changed!